analyzers/src/SonarAnalyzer.CSharp/Extensions/AttributeSyntaxExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Extensions
+{
+    internal static class AttributeSyntaxExtensions
+    {
+        public static bool IsKnownType(this AttributeSyntax attribute, KnownType knownType, SemanticModel semanticModel) =>
+            attribute.Name.GetName().Contains(knownType.ShortNameWithoutAttributeSuffix)
```

Comment:
I've added this overload on `AttributeSyntax` to be able to do a check first at the syntax level, for performance reasons.

Reply:
it looks like "short name without attribute suffix" should be a private method in this class - it's only necessary here, right?

Reply:
I was in a doubt where to add this. It's used only here but it acts and needs only a known type to return the value.

I think I can move it here for now since it makes sense only for attributes.